src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -961,12 +961,38 @@ public override BoundNode VisitPassByCopy(BoundPassByCopy node)
 
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
-            Debug.Assert(!IsConditionalState);
-            VisitRvalue(node.Expression);
-
             bool negated = node.Pattern.IsNegated(out var pattern);
             Debug.Assert(negated == node.IsNegated);
 
+            Debug.Assert(!IsConditionalState);
+            if (VisitPossibleConditionalAccess(node.Expression, out var stateWhenNotNull))
+            {
+                switch (GetTopLevelNullTest(pattern))
+                {
+                    case true:
+                        SetConditionalState(stateWhenNotNull, State);
+                        break;
+                    case false:
+                        SetConditionalState(State, stateWhenNotNull);
+                        break;
+                }
+            }
+            else if (IsConditionalState)
+            {
+                // Note that propagation of conditional state here is intentionally performed only on "simple" constant patterns, not on negated or binary patterns.
```

Comment:
>negated [](start = 137, length = 7)

This feels wrong. `is not true` should have the same behavior as `is false`.

Reply:
Fixed